napari/layers/utils/_state_dict.py

```diff
@@ -0,0 +1,19 @@
+import warnings
+from typing import Any
+
+
+class LayerStateDict(dict[str, Any]):
```

Comment:
Any expertise on the easiest/safest way to subclass `dict`? Seems like `UserDict` used to be the recommended way, but that might not be the case anymore?